api/api-base/src/main/java/com/thoughtworks/go/api/ApiController.java

```diff
@@ -76,6 +78,10 @@ protected void verifyContentType(Request request, Response response) throws IOEx
         }
     }
 
+    protected void setMultpipartUpload(Request req, Response res) {
+        req.raw().setAttribute(MULTIPART_REQ_ATTR, new MultipartConfigElement(System.getProperty("java.io.tmpdir", "tmp")));
```

Comment:
Does this mean that, once the request is served, this directory will not be cleaned up?

Reply:
@arvindsv I had the same question in my head, but I _empirically_ found (at least in my basic trials) that nothing was being written to `java.io.tmpdir` after a multipart request.

FWIW, there's a method on `MultiPartInputStreamParser` that can explicitly delete the parts from disk, but we don't have a handle on that object. There's also a flag for "delete on exit" but same lack of handle problem. So...shrug?

Reply:
Maybe it uses it only if the payload is bigger than its buffer?

Reply:
It’s certainly possible— I’ll try with a larger payload (100mb?). If I see a file not being cleaned, I’ll have an explicit cleanup at the end.